integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/MessageInController.java

```diff
@@ -0,0 +1,193 @@
+            resource = repo.findFirstByOrderByLastUpdateAsc();
+        } else {
+            resource = repo.findFirstByMessagetypeIdOrderByLastUpdateAsc(messagetypeId);
+        }
+
+        if (resource.isPresent()) {
+            return ResponseEntity.ok(resource.get());
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @RequestMapping(value = "/in/messages/asic", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity getZipFromMessage(
+            @RequestParam(value = "conversationId") String conversationId) throws IOException {
+
+        Optional<IncomingConversationResource> resource = Optional.ofNullable(repo.findOne(conversationId));
+        if (!resource.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No conversation with supplied id found.");
+        }
+
+        String filedir = props.getNextbest().getFiledir();
+        if (!filedir.endsWith("/")) {
+            filedir = filedir+"/";
+        }
+        filedir = filedir+conversationId+"/";
+        File file = new File(filedir+props.getNextbest().getAsicfile());
+
+        InputStreamResource isr = new InputStreamResource(new FileInputStream(file));
+        return ResponseEntity.ok()
+                .header("Content-Disposition", "attachement; filename="+props.getNextbest().getAsicfile())
+                .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                .contentLength(file.length())
+                .body(isr);
+    }
+
+    @RequestMapping(value = "/in/messages/file", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity getFileFromMessage(
+            @RequestParam(value = "conversationId") String conversationId,
+            @RequestParam(value = "fileId") Integer fileId) throws IOException {
+
+        Optional<IncomingConversationResource> resource = Optional.ofNullable(repo.findOne(conversationId));
+        if (!resource.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No conversation with supplied id found.");
+        }
+
+        if (!resource.get().getFileRefs().containsKey(fileId)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("No file with given id found for conversation");
+        }
+
+        try {
+            byte[] fileFromAsic = getFileFromAsic(resource.get(), fileId);
+
+            return ResponseEntity.ok()
+                    .header("Content-Disposition", "attachement; filename="+resource.get().getFileRefs().get(fileId))
+                    .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                    .contentLength(fileFromAsic.length)
+                    .body(fileFromAsic);
+        } catch (MessageException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("File with given id not found in archive.");
+        }
+    }
+
+    public byte[] getFileFromAsic(IncomingConversationResource resource, Integer fileId) throws IOException,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, no.difi.meldingsutveksling.noarkexchange.MessageException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)